shared/src/main/scala/mlscript/TyperHelpers.scala

```diff
@@ -79,7 +79,9 @@ abstract class TyperHelpers { self: Typer =>
       cache += tv -> tv
       tv
     case tv: TypeVariable => cache.getOrElse(tv, {
-      val v = freshVar(tv.prov)(tv.level)
+      println(s">>> subst($tv) where cache is $cache")
+      // Inherit name hints so I can identify this variables
```

Comment:
I suppose this comment is no longer valid? You should most definitely not use name hints to base the typing logic on.

Reply:
Removed.

Reply:
Wait, what... we should _always_ keep the name hint! It was a bug that we did not keep it here!

Reply:
So... after this I ended up propagating and introducing name hint in many places! See https://github.com/hkust-taco/mlscript/commit/305e5bcf95898e9b14a4b65b6675a9798cb0abd5.